src/radix_connect/mobile/client.rs

```diff
@@ -1,49 +1,174 @@
-// use super::relay_service::Service as RelayService;
+use std::borrow::BorrowMut;
+
 use super::deep_link_parsing::*;
+use super::relay_service::Service as RelayService;
 use crate::prelude::*;
+use std::sync::RwLock;
 
 /// The Radix Connect Mobile client.
 /// This is the object that will be used by the mobile app to handle interactions sent over Radix Connect Relay.
-// #[derive(uniffi::Object)]
-// pub struct RadixConnectMobile {}
-
-// // Provisional API
-// #[uniffi::export]
-// impl RadixConnectMobile {
-//     // RadixConnectMobile should require a NetworkAntenna and a SecureStorage from the Wallet.
-//     // The internal components, such as RadixConnectRelayService will be created by the RadixConnectMobile.
-//     #[uniffi::constructor]
-//     pub fn new(
-//         _network_antenna: Arc<dyn NetworkAntenna>,
-//         _secure_storage: Arc<dyn SecureStorage>,
-//     ) -> Self {
-//         todo!()
-//     }
-
-//     #[uniffi::method]
-//     pub fn handle_linking_request(
-//         &self,
-//         _request: RadixConnectMobileLinkRequest,
-//     ) -> Result<Url> {
-//         todo!()
-//     }
-
-//     #[uniffi::method]
-//     pub fn handle_dapp_interaction_request(
-//         &self,
-//         _request: RadixConnectMobileDappRequest,
-//     ) -> Result<DappToWalletInteraction> {
-//         todo!()
-//     }
-
-//     #[uniffi::method]
-//     pub fn send_dapp_interaction_response(
-//         &self,
-//         _response: WalletToDappInteractionResponse,
-//     ) -> Result<Url> {
-//         todo!()
-//     }
-// }
+#[derive(uniffi::Object)]
+pub struct RadixConnectMobile {
+    /// The Radix Connect Relay service to be used to communicate with dApps.
+    relay_service: RelayService,
+    /// The secure storage to be used to store session data.
+    secure_storage: WalletClientStorage,
+    /// The new sessions that have been created and are waiting to be validated on dApp side.
+    /// Once the session is validated, it will be moved to the secure storage.
+    /// Validation consists in verifying the origin of the session.
+    new_sessions: RwLock<HashMap<SessionID, Session>>,
+}
+
+// Provisional API
+#[uniffi::export]
+impl RadixConnectMobile {
+    // RadixConnectMobile should require a NetworkAntenna and a SecureStorage from the Wallet.
+    // The internal components, such as RadixConnectRelayService will be created by the RadixConnectMobile.
+    #[uniffi::constructor]
+    pub fn new(
+        network_antenna: Arc<dyn NetworkAntenna>,
+        secure_storage: Arc<dyn SecureStorage>,
+    ) -> Self {
+        Self {
+            relay_service: RelayService::new_with_network_antenna(
+                network_antenna,
+            ),
+            secure_storage: WalletClientStorage::new(secure_storage),
+            new_sessions: RwLock::new(HashMap::new()),
+        }
+    }
+
+    #[uniffi::method]
```

Comment:
The `#[uniffi::method]` is not needed, only used if we wanna rename or set default values.